test/test_server/test_main.cc

```diff
@@ -0,0 +1,85 @@
+#include <array>
+#include <memory>
+#include <fstream>
+#include <common.h>
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+using namespace ::testing;
+
+
+class ServerTest : public ::testing::Test {
+ public:
+  ServerTest() {};
+  virtual ~ServerTest() {};
+
+  virtual void SetUp() {
+  }
+
+  virtual void TearDown() {
+  };
+
+  std::string exec(const char* cmd) {
```

Comment:
Here we are actually launching the `gerbera` binary with the additional options to test its output.